src/Databases/PostgreSQL/DatabaseMaterializedPostgreSQL.cpp

```diff
@@ -133,6 +134,34 @@ void DatabaseMaterializedPostgreSQL::startupTables(ThreadPool & thread_pool, Loa
 }
 
 
+void DatabaseMaterializedPostgreSQL::checkAlterIsPossible(const AlterDatabaseCommands & commands, ContextPtr query_context)
+{
+    auto existing_settings = *settings.get();
+    auto check_setting_change = [&existing_settings, &query_context](const SettingChange & change)
+    {
+        const char * command = change.value.isNull() ? "RESET SETTING" : "MODIFY SETTING";
+        if (change.name == "materialized_postgresql_tables_list" && !query_context->isInternalQuery())
```

Comment:
Please remove this check for `isInternalQuery`, I know it was here before and you only moved it from `applySettingsChanges`, but this check is outdated and I forgot to remove it.